internal/tracker/tracker_test.go

```diff
@@ -158,10 +167,107 @@ task-bundles:
 				_, err = f.WriteString(tt.input)
 				assert.NoError(t, err)
 			}
-			output, err := Track(context.TODO(), tt.urls, "pipeline-bundles", inputFile)
+			output, err := Track(context.TODO(), tt.urls, inputFile, func(ctx context.Context, ref image.ImageReference) ([]string, error) {
+				return []string{"pipeline"}, nil
+			})
 			assert.NoError(t, err)
 			assert.Equal(t, tt.output, string(output))
 		})
 	}
 
 }
+
+func TestTrackerAddKeepsOrder(t *testing.T) {
+	parameters := gopter.DefaultTestParameters()
+
+	// uncomment when test fails and set the seed to the printed value
+	// parameters.Rng.Seed(1234) -
+	arbitraries := arbitrary.DefaultArbitraries()
+
+	nonEmptyIdentifier := gen.Identifier().SuchThat(func(v string) bool {
+		return len(v) > 0
+	})
+	arbitraries.RegisterGen(gen.SliceOf(
+		gen.Struct(reflect.TypeOf(Record{}), map[string]gopter.Gen{
+			"Digest":     gen.RegexMatch("^sha256:[a-f0-9]{64}$"),
+			"Collection": nonEmptyIdentifier,
+			"Tag":        gen.Identifier(),
+			"Repository": nonEmptyIdentifier,
+		})))
+
+	properties := gopter.NewProperties(parameters)
+
+	properties.Property("collections are sorted", arbitraries.ForAll(
+		func(records []Record) bool {
+			tracker := Tracker{}
+			for _, r := range records {
+				tracker.add(r)
+			}
+
+			raw, err := tracker.Output()
+			if err != nil {
+				panic(err)
```

Comment:
Maybe something like `t.Fatal(...)` is better here?

Reply:
Oh maybe it can't because this function is not being called by golang's testing.

Reply:
Yeah, we want to [panic out of gopter tests](https://pkg.go.dev/github.com/leanovate/gopter#example-package-Panic)